src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +95,244 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
+                }
+                else
+                {
+                    paramMergedTypesBuilder.Free();
+                    paramRefCustomModifiersBuilder.Free();
+                    mergedParameterTypes = ParameterTypesWithAnnotations;
+                }
+            }
+
+            if (hasParamChanges || !mergedReturnType.IsSameAs(ReturnTypeWithAnnotations) || mergedReturnRefCustomMods != RefCustomModifiers)
+            {
+                return SubstiteParameterSymbols(mergedReturnType, mergedParameterTypes, mergedReturnRefCustomMods, mergedParamRefCustomModifiers);
+            }
+            else
+            {
+                return this;
+            }
+
+            static ImmutableArray<CustomModifier> mergeCustomModifiers(ImmutableArray<CustomModifier> thisRefCustomModifiers, ImmutableArray<CustomModifier> otherRefCustomModifiers, VarianceKind variance)
+            {
+                Debug.Assert((thisRefCustomModifiers.IsDefault && otherRefCustomModifiers.IsDefault) || thisRefCustomModifiers.Length == otherRefCustomModifiers.Length);
+                if (thisRefCustomModifiers.IsDefaultOrEmpty)
+                {
+                    return thisRefCustomModifiers;
+                }
+                else
+                {
+                    var mergedModifiersBuilder = ArrayBuilder<CustomModifier>.GetInstance(thisRefCustomModifiers.Length);
+                    bool hasCustomModChanges = false;
+                    for (int i = 0; i < thisRefCustomModifiers.Length; i++)
+                    {
+                        var thisModifier = ((CSharpCustomModifier)thisRefCustomModifiers[i]).ModifierSymbol;
+                        var otherModifier = ((CSharpCustomModifier)otherRefCustomModifiers[i]).ModifierSymbol;
+                        var mergedModifier = (NamedTypeSymbol)thisModifier.MergeEquivalentTypes(otherModifier, variance);
+                        if (!ReferenceEquals(thisModifier, mergedModifier))
+                        {
+                            hasCustomModChanges = true;
+                        }
+
+                        mergedModifiersBuilder.Add(thisRefCustomModifiers[i].IsOptional ? CSharpCustomModifier.CreateOptional(mergedModifier) : CSharpCustomModifier.CreateRequired(mergedModifier));
+                    }
+
+                    if (hasCustomModChanges)
+                    {
+                        return mergedModifiersBuilder.ToImmutableAndFree();
+                    }
+                    else
+                    {
+                        return thisRefCustomModifiers;
+                    }
+                }
+            }
+        }
+
+        public FunctionPointerMethodSymbol SetNullabilityForReferenceTypes(Func<TypeWithAnnotations, TypeWithAnnotations> transform)
+        {
+            var transformedReturn = transform(ReturnTypeWithAnnotations);
+            var transformedRefCustomMods = transformRefCustomModifiers(transform, RefCustomModifiers);
```

Comment:
>var transformedRefCustomMods = transformRefCustomModifiers(transform, RefCustomModifiers); [](start = 12, length = 90)

I do not think ever transform custom modifiers, they are not part of the language and meaning of any nullability information in them is undefined. See, for example how ```TypeWithAnnotations.SetUnknownNullabilityForReferenceTypes``` is implemented.